include/zephyr/kernel.h

```diff
@@ -5175,10 +5185,133 @@ __syscall void k_pipe_buffer_flush(struct k_pipe *pipe);
 
 /** @} */
 
+#else /* CONFIG_PIPES */
+
+enum pipe_flags {
+	PIPE_FLAG_OPEN_BIT = 0,
+	PIPE_FLAG_OPEN = BIT(PIPE_FLAG_OPEN_BIT),
+};
+
+struct k_pipe {
+	uint8_t flags;
+	size_t head;
+	size_t tail;
+	size_t count;
```

Comment:
I looked into the `ring_buffer.c ` and it looks like it is two things in one? 
Both an object queue and a ring_buffer.c for bytes. 

The size of the ring_buffer struct alone would almost double the size of the k_pipe object so I opted to not use it. 
(Since I almost had all the logic in place for the ring-buffer already). 

And with your optimization: `head - tail % buffer_size` I've optimized away the count member, but ended up needing that space for the concurrency mechanism for the `flush`/`reset` function.